### PR TITLE
Add /_/health endpoint to of-watchdog

### PR DESCRIPTION
**Pull request by viveksyngh**
Introduce new endpoint `/_/health` to watchdog for health status of
functions which checks for `/tmp/.lock` file

Added tests for healthHandler

Issues: https://github.com/openfaas/faas/issues/547

Signed-off-by: Vivek Singh <vivekkmr45@yahoo.in>

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
  Issue raised by Alex: https://github.com/openfaas/faas/issues/547

## How Has This Been Tested?

This has been tested on Mac by building the watchdog binary. 
```
curl -i http://localhost:8081/_/health
HTTP/1.1 200 OK
Date: Tue, 03 Apr 2018 03:51:24 GMT
Content-Length: 2
Content-Type: text/plain; charset=utf-8

OK%
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: requesthandler_test.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		required := http.StatusMethodNotAllowed
+		if status := rr.Code; status != required {
+			t.Errorf("handler returned wrong status code -  want: %v, got: %v", required, status)
+		}
+	}
+}
+
+func removeLockFile() error {
+	path := filepath.Join(os.TempDir(), ".lock")
+	log.Printf("Removing lock-file : %s\n", path)
```
Comment (alexellis) [inline requesthandler_test.go]: We shouldn't encourage logging messages in tests.

**Comment by alexellis** [comment]
I think this looks fine. @johnmccabe can you comment?

File: requesthandler_test.go (hunk 1 above)
Comment (johnmccabe) [reply requesthandler_test.go]: There are methods in the `testing` package for this. `t.Log` and `t.Logf` which only get printed when a test fails or the `-test.v` flag is set. See [here](https://golang.org/pkg/testing/#T.Log).

File: main.go [hunk 2]
```diff
@@ -223,3 +224,29 @@ func makeHTTPRequestHandler(watchdogConfig config.WatchdogConfig) func(http.Resp
...
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
+
+func makeHealthHandler() func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		switch r.Method {
+		case "GET":
```
Comment (johnmccabe) [inline main.go]: use `http.MethodGet` here

File: main.go [hunk 3]
```diff
@@ -223,3 +224,29 @@ func makeHTTPRequestHandler(watchdogConfig config.WatchdogConfig) func(http.Resp
...
+
+func lockFilePresent() bool {
+	path := filepath.Join(os.TempDir(), ".lock")
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
+
+func makeHealthHandler() func(w http.ResponseWriter, r *http.Request) {
```
Comment (johnmccabe) [inline main.go]: Should this not be the same as the old watchdog implementation? The `acceptingConnections` check is missing here and from the `lockFilePresent()`. Why is that no longer needed?

Comment (BurtonR) [reply main.go]: The `acceptingConnections` check is part of the graceful shutdown that's currently only in the original watchdog. There's an issue to port it over here: #10 

I'm starting to work on that. Hoping to have a PR tonight (US time)

File: requesthandler_test.go (hunk 1 above)
Comment (viveksyngh) [reply requesthandler_test.go]: I have removed the log statement.

File: main.go (hunk 2 above)
Comment (viveksyngh) [reply main.go]: Agree .. I have updated this.

**Comment by johnmccabe** [comment]
Lgtm 👍

**Pull request merged**